.editorconfig

```diff
@@ -0,0 +1,35 @@
+# https://editorconfig.org/
+# specification: https://editorconfig-specification.readthedocs.io/en/latest/
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+[*.{c,h,hh,cc,cci}]
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+
+[Makefile]
+indent_style = tab
+
+[*.am]
+indent_style = tab
+
+[configure.ac]
+indent_style = space
+indent_size = 2
+
+[*.awk]
+indent_style = space
+indent_size = 4
+
+[*.{pl,pl.in,pm}]
+indent_style = tab
```

Comment:
> > I hand-checked samples for each category, and built the rules according what I saw in these files.
> 
> FWIW, searching for four spaces in the beginning of a line shows many `indent_style = tab` violations for Perl files:
> 
> ```shell
> git ls-files | egrep '[.](pl|pm)(.in)?$' | xargs grep '^    ' | wc -l
> 863 # lines
> ```
> 
> ```shell
> git ls-files | egrep '[.](pl|pm)(.in)?$' | xargs grep -m1 '^    ' | wc -l
> 33 # files
> ```
> 
> In fact, the number of Perl files using spaces (33) is about the same as the number of Perl files using tabs (37)! There is a large overlap between the two sets -- the total number of Perl files is 44 -- but that does not matter here. This finding makes me wonder if we should use `indent_style=space` for Perl files instead: Since the current sources are inconsistent, we should use a better default -- spaces -- for the same reasons we are using spaces for C++ files _and_ because we are already using spaces for C++ and other file types.

I'm fine with the idea. I'd like to head @yadij 's voice before implementing the change.


> A separate PR replacing tabs with spaces in all Perl files would be very welcomed. If you do not want to volunteer such a PR, then we should probably disable the indentation rules for now:

I can do it, no problem.

Reply:
FTR: #660 reformats Perl files.